opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryNoRecordEventsSpanImpl.java

```diff
@@ -0,0 +1,183 @@
+  }
+
+  @Override
+  public void putAttribute(String key, AttributeValue value) {
+    Preconditions.checkNotNull(key, "key");
+    Preconditions.checkNotNull(value, "value");
+  }
+
+  @Override
+  public void putAttributes(Map<String, AttributeValue> attributes) {
+    Preconditions.checkNotNull(attributes, "attributes");
+  }
+
+  @Override
+  public void addMessageEvent(MessageEvent messageEvent) {
+    Preconditions.checkNotNull(messageEvent, "messageEvent");
+  }
+
+  @Override
+  public void addLink(Link link) {
+    Preconditions.checkNotNull(link, "link");
+  }
+
+  @Override
+  public void setStatus(Status status) {
+    Preconditions.checkNotNull(status, "status");
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span setStatus(StatusCode canonicalCode) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span setStatus(StatusCode canonicalCode, String description) {
+    return null;
+  }
+
+  @Override
+  public void end(EndSpanOptions options) {
+    Preconditions.checkNotNull(options, "options");
+  }
+
+  @Override
+  public void end(long timestamp, TimeUnit unit) {
+    // do nothing
+  }
+
+  private OpenTelemetryNoRecordEventsSpanImpl(SpanContext context) {
+    super(context, NOT_RECORD_EVENTS_SPAN_OPTIONS);
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span setAttribute(String key, @Nonnull String value) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span setAttribute(String key, long value) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span setAttribute(String key, double value) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span setAttribute(String key, boolean value) {
+    return null;
+  }
+
+  @Override
+  public <T> io.opentelemetry.api.trace.Span setAttribute(AttributeKey<T> key, @Nonnull T value) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span addEvent(String name) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span addEvent(String name, long timestamp, TimeUnit unit) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span addEvent(String name, Attributes attributes) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span addEvent(
+      String name, Attributes attributes, long timestamp, TimeUnit unit) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span recordException(Throwable exception) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span recordException(
+      Throwable exception, Attributes additionalAttributes) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span updateName(String name) {
+    return null;
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.SpanContext getSpanContext() {
+    return io.opentelemetry.api.trace.SpanContext.getInvalid();
```

Comment:
Shouldn't it be the mapped context, not invalid? Is `SpanContext` guaranteed to be invalid in the constructor? I'm guessing no since in that case we wouldn't even need a constructor parameter for it.